include/rocksdb/file_system.h

```diff
@@ -749,13 +756,33 @@ class FSWritableFile {
   //
   // PositionedAppend() requires aligned buffer to be passed in. The alignment
   // required is queried via GetRequiredBufferAlignment()
+
+  // Append data with verification information
+  // Note that this API change is experimental and it might be changed in
+  // the future. Currently, RocksDB does not use this API.
+  virtual IOStatus Append(const Slice& data, const IOOptions& options,
+                          IODebugContext* dbg,
+                          const DataVerificationInfo& /* verification_info */) {
```

Comment:
I seems a little odd to me for DataVerificationInfo to come after IOOptions and IODebugContext. To me, this information seems more "important" than IOOptions and IODebugContext so should come before them. Of course DataVerificationInfo is less "important" than data (and offset). For example, the parameter orderings would be weird to me if we added another parameter onto the end of both Append overloads.

What do others think?

Reply:
I agree that the order is suboptimal. dbg does feel should come later than DataVerificationInfo. Not sure IOOptions. RocksDB public functions tend to put options before data arguments, e.g. DB::Get() and DB::Put().

Reply:
Yes, it make more scene. How about between IOOotions and IOdebugContext? @pdillinger @siying